File: mybatis-plus-join-test/test-join/src/main/java/com/github/yulichang/test/join/entity/OrderDO.java
package com.github.yulichang.test.join.entity;

import com.baomidou.mybatisplus.annotation.OrderBy;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.ToString;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@ToString
@Accessors(chain = true)
@TableName("order_t")
public class OrderDO implements Serializable {

    @TableId
    private Integer id;

    private Integer userId;

    @OrderBy
    private String name;

    @TableField(exist = false)
    private String userName;
}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/entity/OrderDO.kt
package com.github.yulichang.test.kt.entity

import com.baomidou.mybatisplus.annotation.OrderBy
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import java.io.Serializable


@TableName("order_t")
class OrderDO : Serializable {
    @TableId
    val id: Int? = null
    val userId: Int? = null

    @OrderBy
    val name: String? = null

    @TableField(exist = false)
    val userName: String? = null
}
